lib/mllp/receiver.ex

```diff
@@ -62,10 +84,71 @@ defmodule MLLP.Receiver do
   @spec stop(any) :: :ok | {:error, :not_found}
   def stop(port) do
     receiver_id = get_receiver_id_by_port(port)
-
     :ok = :ranch.stop_listener(receiver_id)
   end
 
+  @doc """
+  A function which can be used to embed an MLLP.Receiver under Elixir v1.5+ supervisors.
+  Unlike `start/1`, `start/2`, or `start/3` this function takes two additional options : `ref` and `transport_opts`.
+  Note that if a `ref` option is not supplied a reference will be created for you using `make_ref/0`.
+      children = [{MLLP.Receiver, [
+          ref: MyRef,
+          port: 4090,
+          dispatcher: MyDispatcher,
+          packet_framer: MyPacketFramer,
+          transport_opts: %{num_acceptors: 25, max_connections: 20_000}
+        ]}
+      ]
+      Supervisor.init(children, strategy: :one_for_one)
+  See [Options](#module-options) for details on required and optiomal parameters.
+  ## Examples
+      iex(1)> opts = [ref: MyRef, port: 4090, dispatcher: MLLP.DefaultDispatcher, packet_framer: MLLP.DefaultPacketFramer]
+      [
+        ref: MyRef,
+        port: 4090,
+        dispatcher: MLLP.DefaultDispatcher,
+        packet_framer: MLLP.DefaultPacketFramer
+      ]
+      iex(2)> MLLP.Receiver.child_spec(opts)
+      %{
+        id: {:ranch_listener_sup, MyRef},
+        modules: [:ranch_listener_sup],
+        restart: :permanent,
+        shutdown: :infinity,
+        start: {:ranch_listener_sup, :start_link,
+         [
+          MyRef,
+          :ranch_tcp,
+          %{socket_opts: [port: 4090]},
+          MLLP.Receiver,
+          [packet_framer_module: MLLP.DefaultPacketFramer, dispatcher_module: MLLP.DefaultDispatcher]
+        ]},
+        type: :supervisor
+      }
+  """
+  @spec child_spec(options()) :: Superviosr.child_spec()
```

Comment:
There is type in Supervisor. How did dialyzer pass?🤔

Reply:
Interesting! Looking further before I fix this up.

Reply:
To note, it's because our dialyzer setup  is strict enough right now. We need to add more flags to the config, but only added `:unknown` for now, as adding the rest will result in a lot of warnings and errors and changes to a lot of files. Doing this should be the next PR imo.